scripts/quality_test.py

```
@@ -2,171 +2,188 @@
+                return None
+            else:
+                return report_file.read()
+    else:
+        file_not_found_message = f"FAILURE: The following log file could not be found: {filename}"
+        fail_quality(report_name, file_not_found_message)
 
-    # Make sure the metrics subdirectory exists
-    Env.METRICS_DIR.makedirs_p()
 
-    if not report.exists():
-        sh(f'pycodestyle . | tee {report} -a')
+def _get_count_from_last_line(filename, file_type):
+    """
+    This will return the number in the last line of a file.
+    It is returning only the value (as a floating number).
+    """
+    report_contents = _get_report_contents(filename, file_type, last_line_only=True)
+    if report_contents is None:
+        return 0
 
-    violations_list = _pep8_violations(report)
+    last_line = report_contents.strip()
+    # Example of the last line of a compact-formatted eslint report (for example): "62829 problems"
+    regex = r'^\d+'
 
-    return len(violations_list), violations_list
+    try:
+        return float(re.search(regex, last_line).group(0))
+    # An AttributeError will occur if the regex finds no matches.
+    # A ValueError will occur if the returned regex cannot be cast as a float.
+    except (AttributeError, ValueError):
+        return None
 
 
-def _pep8_violations(report_file):
+def _get_stylelint_violations():
     """
-    Returns the list of all PEP 8 violations in the given report_file.
+    Returns the number of Stylelint violations.
     """
-    with open(report_file) as f:
-        return f.readlines()
+    REPO_ROOT = repo_root()
+    REPORT_DIR = REPO_ROOT / 'reports'
+    stylelint_report_dir = (REPORT_DIR / "stylelint")
+    stylelint_report = stylelint_report_dir / "stylelint.report"
+    _prepare_report_dir(stylelint_report_dir)
 
+    command = [
+        'node', 'node_modules/stylelint',
+        '*scss_files',
+        '--custom-formatter', 'stylelint-formatter-pretty/index.js'
+    ]
```

This command was silently passing because:

* `node_modules/stylelint` is a directory, not a binary. The binary is at `node_modules/.bin/stylelint`. Running `node` on a directory does nothing.
* `*scss_files` matches nothing. The correct pattern is `**/*.scss`

Regardless, though, the Paver version of this command doesn't work on master either. It has been silently failing with `/bin/sh: 1: stylelint: not found` for potentially several years now. There are now 24k+ sass style violations on master. Fixing all those violations would not be a good use of our time, especially since we intend to delete all the Sass code in edx-platform.

So, you can just remove all the `stylelint` checks.